### PR TITLE
Show alert when attempting to take photo after disabling camera permission. #1346

### PR DESCRIPTION
**Pull request by jaropawlak**
### First time contributor checklist

- [x] I have read the [README](https://github.com/WhisperSystems/Signal-iOS/blob/master/README.md) and [CONTRIBUTING](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md) documents
- [x] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)

### Contributor checklist

- [x] I'm following the [code, UI and style conventions](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md#code-conventions)
- [x] My commits are rebased on the latest master branch
- [x] My commits are in nice logical chunks
- [x] My contribution is fully baked and is ready to be merged as is
- [x] I have tested my contribution on these devices:
 * iPhone 5, iOS 10
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

- - - - - - - - - -

### Description

…ssion. #1346

### CODE REVIEW

File: Signal/src/view controllers/MessagesViewController.m [hunk 1]
```diff
@@ -1475,15 +1475,30 @@ - (void)takePictureOrVideo {
...
+        
+        [alert addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"OK",nil) style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+            [[UIApplication sharedApplication] openURL:[NSURL URLWithString:UIApplicationOpenSettingsURLString]];
+        }]];
+        
+        [alert addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"TXT_CANCEL_TITLE",nil) style:UIAlertActionStyleCancel handler:^(UIAlertAction * _Nonnull action) {
+        }]];
+        
+        [self presentViewController:alert animated:YES completion:[UIUtil modalCompletionBlock]];
+        } else if (status == AVAuthorizationStatusNotDetermined || status == AVAuthorizationStatusAuthorized) {
```
Comment (michaelkirk) [inline Signal/src/view controllers/MessagesViewController.m]: nit: outdent.

File: Signal/src/view controllers/MessagesViewController.m [hunk 2]
```diff
@@ -1475,15 +1475,30 @@ - (void)takePictureOrVideo {
...
+        
+        UIImagePickerController *picker = [[UIImagePickerController alloc] init];
+        picker.sourceType = UIImagePickerControllerSourceTypeCamera;
+        picker.mediaTypes = @[ (__bridge NSString *)kUTTypeImage, (__bridge NSString *)kUTTypeMovie ];
+        picker.allowsEditing = NO;
+        picker.delegate = self;
+        [self presentViewController:picker animated:YES completion:[UIUtil modalCompletionBlock]];
+    }
```
Comment (michaelkirk) [inline Signal/src/view controllers/MessagesViewController.m]: We should DDLogError " unknown AVAuthorizationStatus: %d" in a final else branch.

File: Signal/src/view controllers/MessagesViewController.m (hunk 1 above)
Comment (jaropawlak) [inline Signal/src/view controllers/MessagesViewController.m]: done

File: Signal/src/view controllers/MessagesViewController.m (hunk 2 above)
Comment (jaropawlak) [inline Signal/src/view controllers/MessagesViewController.m]: done (with %ld as that was what Xcode has suggested)

**Comment by jaropawlak** [comment]
Requested changes were made commits were squashed

**Comment by jaropawlak** [comment]
what's going on with cla-server? I filled the cla third time and status is still "cla-server/validation — Waiting for status to be reported"

**Comment by michaelkirk** [comment]
> what's going on with cla-server? I filled the cla third time and status is still "cla-server/validation — Waiting for status to be reported"

Sometimes I fail it to. It's completely opaque to me. Can you please try closing this pull request and opening another?

**Pull request closed without merging**